core/arch/riscv/mm/core_mmu_arch.c

```diff
@@ -628,16 +628,16 @@ static void set_user_va_idx(struct mmu_partition *prtn)
 			return;
 		}
 	}
-	if (user_va_idx < 0)
-		panic();
+
+	assert(core_mmu_user_va_range_is_defined());
```

Comment:
This replaces a always embedded test with a debug config assertion. Looks fine but I think it should be stated in the commit message.

Reply:
Upon reflection, I should keep the test unchanged.